### DISCUSSION TITLE
The code snippet in Wiki doesn't compile

### DISCUSSION THREAD

**Discussion by lenguyenthanh** [j-mie6/Parsley#188] [General]
Hi, thanks for the wiki, I'm reading it (slowly) and it really helps me. 

I see that the code snippet in [chain](https://github.com/j-mie6/Parsley/wiki/Building-Expression-Parsers#using-exprchain) doesn't compile. Here the fix:

`lazy val expr: Parsley[Int] = chain.left1(term, ('+' #> add <|> '-' #> sub).map(f => (x, y) => f(y)(x)))`

I'm not sure it is intended (for making thing simpler) so I post it here. Hope that this is the right place.

**Comment by j-mie6** [comment]
Are you sure, it compiles fine in Scastie:

![image](https://github.com/j-mie6/parsley/assets/5148976/336d0d08-94e2-4ca6-b6c8-6068c173d72d)

admittedly, I missed a `lazy` on term (@roccojiang you'll want to add this to your branch, I've also added one to `atom` just because!)

**Reply by j-mie6** [reply]
`add` and `sub` are both uncurried here

**Reply by j-mie6** [reply]
Ah, right you're using the old `add` and `sub` functions. I should point out more clearly they've been uncurried!

**Reply by lenguyenthanh** [reply]
ah, sorry, I missed that changes. I copy/paste the parser part, without update `add` and `sub`. thanks!

**Reply by j-mie6** [reply]
helped me spot the missing lazy anyway 😄
